### PR TITLE
Add inline mappings and simplify configuration

### PR DESCRIPTION
**Pull request by jnraine**
This changes the interface of the gem, removing the need for external mappings and minimizing the configuration difference between integer type and normal polymorphic associations.

```ruby
class Picture < ActiveRecord::Base
  include PolymorphicIntegerType::Extensions

  belongs_to :imageable, polymorphic: {1 => "Employee", 2 => "Product"}
end

class Employee < ActiveRecord::Base
  include PolymorphicIntegerType::Extensions

  has_many :pictures, as: :imageable
end

class Product < ActiveRecord::Base
  include PolymorphicIntegerType::Extensions

  has_many :pictures, as: :imageable
end

Picture.imageable_type_mapping
# => {1 => "Employee", 2 => "Product"}
```

No external mapping necessary.

This maintains compatibility with external mappings and usage of the `integer_type:` option so existing code will continue to work.

### CODE REVIEW

File: lib/polymorphic_integer_type/extensions.rb [hunk 1]
```diff
@@ -7,10 +7,22 @@ def belongs_to(name, scope = nil, options = {})
...
         super
-        if options[:polymorphic] && integer_type
-          mapping = PolymorphicIntegerType::Mapping[name]
+        if options[:polymorphic] && (integer_type || options[:polymorphic].is_a?(Hash))
+          mapping =
+            case integer_type
+            when true then PolymorphicIntegerType::Mapping[name]
+            when nil then options[:polymorphic]
+            else
+              integer_type
```
Comment (doliveirakn) [inline lib/polymorphic_integer_type/extensions.rb]: when would we have this default?

File: lib/polymorphic_integer_type/extensions.rb [hunk 2]
```diff
@@ -59,6 +83,22 @@ def remove_type_and_establish_mapping(name, options, scope)
         end
       end
 
+      def retrieve_polymorphic_type_mapping(polymorphic_type:, class_name:)
+        return if polymorphic_type == nil
+
+        belongs_to_class = begin
+          class_name.constantize
```
Comment (doliveirakn) [inline lib/polymorphic_integer_type/extensions.rb]: good ol' rails. there is a method for this. `safe_constantize`

File: lib/polymorphic_integer_type/extensions.rb [hunk 3]
```diff
@@ -59,6 +83,22 @@ def remove_type_and_establish_mapping(name, options, scope)
         end
       end
 
+      def retrieve_polymorphic_type_mapping(polymorphic_type:, class_name:)
+        return if polymorphic_type == nil
```
Comment (doliveirakn) [inline lib/polymorphic_integer_type/extensions.rb]: `.nil?`?

File: lib/polymorphic_integer_type/extensions.rb (hunk 1 above)
Comment (jnraine) [reply lib/polymorphic_integer_type/extensions.rb]: Good catch. Originally, the mapping was passed to `integer_type:`. We can drop this case.

File: lib/polymorphic_integer_type/extensions.rb (hunk 3 above)
Comment (jnraine) [reply lib/polymorphic_integer_type/extensions.rb]: 😄  The Sandi Metz training instilled the idea of equality checks over all other conditionals. For this case, I don't think entirely necessary. I'll use `nil?` instead.

**Pull request merged**